### ISSUE TITLE
non-valid dtypes

### ISSUE THREAD

**Issue by lzehl** [G-Node/python-odml#39]
Hello, an odml file with a non-valid dtype of a value can currently be saved, but not reloaded again.
This is relatively dangerous... I suggest that an error is directly raised while generating the value, before saving the odml file. I'll have a look and see if I can make a pull request.

Best Lyuba

**Comment by jgrewe** [comment]
Hi Lyuba, 
this is indeed critical and as far as I understand due to changes in the enum package. It would be great if you could patch this. Maybe we can check at the point of writing whether the type info is valid and  handle this transparently. Raising and error is maybe not a good idea because it is too late to fix it easily for the user.

**Comment by jgrewe** [comment]
Example code that leads to the issue:
````python
doc = odml.Document()
doc.append(odml.Section('sec'))
doc['sec'].append(odml.Property('prop', value=odml.Value(42, dtype=odml.DType.int)))
odml.tools.xmlparser.XMLWriter(doc).write_file(save_to)
redoc = odml.tools.xmlparser.load('testDTypeEnum34.odml')
````

Code that works:
````python
doc = odml.Document()
doc.append(odml.Section('sec'))
doc['sec'].append(odml.Property('prop', value=odml.Value(42)))
odml.tools.xmlparser.XMLWriter(doc).write_file(save_to)
redoc = odml.tools.xmlparser.load('testDTypeEnum34.odml')
````
In the latter case the type of the value is not explicitly stated and will be guessed by the library. This works just fine for all basic data types.

**Comment by lzehl** [comment]
Hi, I'll need to have a closer look at the code again, but I think the dtype is already checked for validity, at least when a file is reloaded (that is causing the error). I will have a look today why the validation is not done while creating the value or saving the file. This problem is of course connected to the enum-vs-enum34 issue, but if a dtype is given specifically by the user, but with a typo, this causes the same problem... For now, we implemented a work around to this problem already, similar to what you suggested. Thanks for having a look. I'll write again if I come up with something.

**Comment by jgrewe** [comment]
On 09/25/2015 09:25 AM, Lyuba Zehl wrote:
> Hi, I'll need to have a closer look at the code again, but I think the
> dtype is already checked for validity, at least when a file is reloaded
> (that is causing the error). I will have a look today why the validation
> is not done while creating the value or saving the file. This problem is
> of course connected to the enum-vs-enum34 issue, but if a dtype is given
> specifically by the user, but with a typo, this causes the same
> problem... For now, we implemented a work around to this problem
> already, similar to what you suggested. Thanks for having a look. I'll
> write again if I come up with something.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/G-Node/python-odml/issues/39#issuecomment-143148453>.
> 

Thanks a lot!!

**Comment by jgrewe** [comment]
@lzehl could you confirm if the bug has indeed been fixed on your systems? It is working fine with my installation but who knows...

**Comment by jgrewe** [comment]
solved via  pr #42
